lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -26,279 +26,189 @@ def _config_str_to_list(self, list_str):
-        if self.ws_narratives is None:
-            self.ws_narratives = self.metrics_dbi.list_ws_narratives()
-        if self.client_groups is None:
-            self.client_groups = self.get_client_groups_from_cat(token)
+    def _parse_app_id(self, exec_task):
+        app_id = ''
+        if 'app_id' in exec_task['job_input']:
+            app_id = exec_task['job_input']['app_id'].replace('.', '/')
 
-        # 1. update users
-        action_result1 = self.update_user_info(requesting_user, params, token)
+        return app_id
 
-        # 2. update activities
-        action_result2 = self.update_daily_activities(requesting_user, params, token)
+    def _parse_method(self, exec_task):
+        method_id = ''
+        if 'method' in exec_task['job_input']:
+            method_id = exec_task['job_input']['method'].replace('/', '.')
 
-        # 3. update narratives
-        action_result3 = self.update_narratives(requesting_user, params, token)
+        return method_id
 
-        return {'metrics_result': {'user_updates': action_result1,
-                                   'activity_updates': action_result2,
-                                   'narrative_updates': action_result3}}
+    def _map_narrative(self, wsid, ws_narrs):
+        """
+        get the narrative name and version
+        """
+        n_name = ''
+        n_obj = '0'
+        for ws in ws_narrs:
+            if str(ws['workspace_id']) == str(wsid):
+                n_name = ws['name']
+                w_meta = ws['meta']
+                for w_m in w_meta:
+                    if w_m['k'] == 'narrative':
+                        n_obj = w_m['v']
+                    if w_m['k'] == 'narrative_nice_name':
+                        n_name = w_m['v']
+                break
+        return (n_name, n_obj)
 
-    def update_user_info(self, requesting_user, params, token):
+    def _update_user_info(self, requesting_user, params, token):
         """
         update user info
         If match not found, insert that record as new.
         """
-        if not self.is_metrics_admin(requesting_user):
-            raise ValueError('You do not have permission to invoke this action.')
+        if not self._is_metrics_admin(requesting_user):
+            raise ValueError('You do not have permission to '
+                             'invoke this action.')
 
-        params = self.process_parameters(params)
-        # TODO: set the param['minTime'] and param['maxTime'] to a given time window,
-        # such as most current 24 hours instead of 48 hours as for others.
+        params = self._process_parameters(params)
         auth2_ret = self.metrics_dbi.aggr_user_details(
             params['user_ids'], params['minTime'], params['maxTime'])
-        updData = 0
+        upDated = 0
+        upSerted = 0
         if len(auth2_ret) == 0:
             print("No user records returned for update!")
-            return updData
+            return (0, 0)
```

Comment:
Add a test for this case.

Should it return 0 instead of (0, 0)?

Reply:
added test to return 0